tests/rbd/test_rbd_encryption.py

```diff
@@ -0,0 +1,314 @@
+            flag = 1
+
+    return flag
+
+
+def test_rbd_encryption(rbd, pool_type, format, **kw):
+    """
+    Perform luks encryption on the given rbd pool, image and its clone and verify data integrity
+    Args:
+        rbd: rbd object
+        pool_type: pool type (ec_pool_config or rep_pool_config)
+        format: encryption format of parent and clone Ex: luks1,luks2
+        **kw: Test data
+    """
+    [parent_encryption_type, clone_encryption_type] = format.split(",")
+    pool = kw["config"][pool_type]["pool"]
+    image = f"{parent_encryption_type}_{rbd.random_string()}"
+    size = "1G"
+    kw["config"][pool_type]["size"] = size
+    snap = f"{clone_encryption_type}_snap_{rbd.random_string()}"
+    clone = f"{clone_encryption_type}_clone_{rbd.random_string()}"
+    image_spec = f"{pool}/{image}"
+    clone_spec = f"{pool}/{clone}"
+    snap_spec = f"{pool}/{image}@{snap}"
+    encryption_config = list()
+    parent_file_path = f"/tmp/{image}_dir/{image}_file"
+    clone_file_path = f"/tmp/{clone}_dir/{image}_file"
+
+    try:
+        io_config = {
+            "file_size": "100M",
+            "file_path": [parent_file_path],
+            "image_spec": [f"{image_spec}"],
+            "operations": {
+                "fs": "ext4",
+                "io": True,
+                "mount": True,
+                "nounmap": True,
+                "device_map": True,
+            },
+            "runtime": 30,
+        }
+
+        rbd.create_image(pool, image, size)
+        if parent_encryption_type.startswith("luks"):
+            log.info(
+                f"Applying luks encryption of type {parent_encryption_type} on {image_spec}"
+            )
+            passphrase = f"{parent_encryption_type}_passphrase.bin"
+            rbd.create_passphrase_file(passphrase)
+            if rbd.encrypt(image_spec, parent_encryption_type, passphrase):
+                log.error(
+                    f"Apply RBD encryption {parent_encryption_type} failed on {image_spec}"
+                )
+
+            encryption_passphrase = {
+                "passphrase": [{"encryption-passphrase-file": passphrase}]
+            }
+
+            if rbd.image_resize(pool, image, size, **encryption_passphrase):
+                log.error(
+                    "Image resize to compensate overhead due to "
+                    f"{parent_encryption_type} header failed for {image_spec}"
+                )
+                return 1
+
+            encryption_config.append({parent_encryption_type: passphrase})
+        else:
+            log.info("Parent image is not encrypted")
+
+        io_config["encryption_config"] = encryption_config
+        kw["config"].update(io_config)
+        kw.update({"rbd_obj": rbd, "do_not_create_image": True})
+        if krbd_io_handler(**kw):
+            log.error(f"Map, mount and run IOs failed for {image_spec}")
+            return 1
+
+        # Sleep for sometime after running IOs before creating snap
+        sleep(60)
```

Comment:
if I/O is large does this sleep value is ok?

Reply:
As of now IO size is fixed at 100M, we can modify this when we are automating tier-3 scenarios for encryption which do not exist even in polarion as of not. 
Can be taken up as part of different PR IMO